packages/serverless/src/model/abstract/document.ts

```diff
@@ -1,24 +1,75 @@
 /* eslint-disable no-await-in-loop */
 // eslint-disable-next-line max-classes-per-file
+import {
+  TDocumentField,
+  TDocumentRecord,
+  TProvableTypeString,
+} from '@zkdb/common';
 import { DB, ModelBasic, ModelCollection, ModelDatabase } from '@zkdb/storage';
 import { randomUUID } from 'crypto';
-import { ClientSession, Filter, OptionalId } from 'mongodb';
+import { ClientSession, Filter, Long, OptionalId } from 'mongodb';
 import logger from '../../helper/logger.js';
 import { getCurrentTime } from 'helper/common.js';
 import { TPickOptional } from '@orochi-network/framework';
-import { TDocumentField, TDocumentRecord } from '@zkdb/common';
 
 // TODO: the naming maybe confusing with TDocumentRecord from common
 export type IDocumentRecord = TPickOptional<
   TDocumentRecord,
   'previousObjectId'
 >;
 
+/** Database-serialized version of a document record. */
+export type TDocumentRecordSerialized = TPickOptional<
```

Comment:
This type must be in `@zkdb/common`

Reply:
this defines the serialization mapping between javascript types and mongodb types. because serverless is the only thing that talks to the database so I think leaving it here is ok